### PR TITLE
[ORC] replace threadpool with createThread in the runnableExamples

### PR DESCRIPTION
**Pull request by ringabout**
`threadpool` doesn't work with ORC, so I replace threadpool with createThread.

### CODE REVIEW

File: lib/pure/random.nim [hunk 1]
```diff
@@ -178,24 +178,31 @@ proc skipRandomNumbers*(s: var Rand) =
...
+    import std/random
 
-    const spawns = 4
     const numbers = 100000
 
-    proc randomSum(r: Rand): int =
-      var r = r
+    var
+      thr: array[0..3, Thread[(Rand, int)]]
+      vals: array[0..3, int]
```
Comment (ringabout) [inline lib/pure/random.nim:187]: Threads can only write to different memory locations, there is no need to use locks.

File: lib/pure/random.nim [hunk 2]
```diff
@@ -178,24 +178,31 @@ proc skipRandomNumbers*(s: var Rand) =
...
+      createThread(thr[i], randomSum, (r, i))
       r.skipRandomNumbers()
 
+    joinThreads(thr)
+
     for val in vals:
-      doAssert abs(^val - numbers * 5) / numbers < 0.1
+      doAssert abs(val - numbers * 5) / numbers < 0.1
+
+    doAssert vals == [501737, 497901, 500683, 500157]
```
Comment (ringabout) [inline lib/pure/random.nim:204]: The old example gives the same result.

**Review by Varriount — APPROVED** [review]

**Comment by c-blake** [comment]
As written, this cache thrashes which I think sets a bad example.  E.g., on a 16-core CPU (changing `0..3` to `0..15` and `numbers` to `1_000_000_000` to make it take long enough), I find that code with this patch runs about 14X faster:
```diff
 proc randomSum(params: tuple[r: Rand, index: int]) {.thread.} =
   var r = params.r
-  for i in 1..numbers:
-    vals[params.index] += r.rand(0..10)
+  var s = 0
+  for i in 1..numbers: s += r.rand(0..10)
+  vals[params.index] = s
```

**Comment by ringabout** [comment]
@c-blake  Hello, thanks for your patch. Would you mind opening a PR or you would like me to push this patch?

**Comment by c-blake** [comment]
You can just patch it.  You may want to put a comment after `var s = 0`, say `# avoid cache thrash`.  I think there are a few instances of this around.  As you do your ORC default work, you may want to keep a watchful eye.  People often start with working examples for their own code (or, LOL, for another runnable example).

**Comment by ringabout** [comment]
=> https://github.com/nim-lang/Nim/pull/20129

**Pull request merged**